src/main/java/edu/berkeley/ground/dao/models/cassandra/CassandraEdgeFactory.java

```diff
@@ -88,6 +88,20 @@ public Edge create(String name,
                      long fromNodeId,
                      long toNodeId,
                      Map<String, Tag> tags) throws GroundException {
+
+    Edge edge = null;
+    try {
+      edge = this.retrieveFromDatabase(sourceKey);
+    } catch (GroundException e) {
+      if (!e.getMessage().contains("No Edge found")) {
```

Comment:
Checking messages in exception is fragile and not refactor-friendly. See an alternative approach here: https://github.com/vsreekanti/ground/pull/1